### PR TITLE
[Bots] Cleanup BotDatabase::LoadBuffs

### PR DESCRIPTION
**Pull request by Aeadoin**
Added Checks to initialize all buff slots (mimics Client LoadBuffs method)
Cleaned up all atoi conversions, change to Strings::ToInt and atoul where applicable.

### CODE REVIEW

File: zone/bot_database.cpp [hunk 1]
```diff
@@ -663,45 +664,60 @@ bool BotDatabase::LoadBuffs(Bot* bot_inst)
...
-		" WHERE `bot_id` = '%u'",
+		" WHERE `bot_id` = '{}'",
 		bot_inst->GetBotID()
 	);
 	auto results = database.QueryDatabase(query);
-	if (!results.Success())
+	if (!results.Success()) {
 		return false;
-	if (!results.RowCount())
+	}
```
Comment (Kinglykrab) [inline zone/bot_database.cpp:673]: Newline between here?

File: zone/bot_database.cpp [hunk 2]
```diff
@@ -663,45 +664,60 @@ bool BotDatabase::LoadBuffs(Bot* bot_inst)
 		" `extra_di_chance`,"
 		" `instrument_mod`"
 		" FROM `bot_buffs`"
-		" WHERE `bot_id` = '%u'",
+		" WHERE `bot_id` = '{}'",
```
Comment (Kinglykrab) [inline zone/bot_database.cpp:667]: `'{}'` is only necessary for strings, can just do `{}`.

**Review by Kinglykrab — COMMENTED** [review]
May be worth converting this to load the data from repositories. Not now, but in the future.

File: zone/bot_database.cpp [hunk 3]
```diff
@@ -663,45 +664,60 @@ bool BotDatabase::LoadBuffs(Bot* bot_inst)
...
+			bot_buffs[buff_count].counters = atoul(row[5]);
+		}
+		else if (CalculateCurseCounters(bot_buffs[buff_count].spellid) > 0) {
+			bot_buffs[buff_count].counters = atoul(row[6]);
+		}
+		else if (CalculateCorruptionCounters(bot_buffs[buff_count].spellid) > 0) {
+			bot_buffs[buff_count].counters = atoul(row[7]);
+		} else {
+			bot_buffs[buff_count].counters = 0;
+		}
```
Comment (Kinglykrab) [inline zone/bot_database.cpp:709]: Not sure our standard, but I usually try to do this.
```cpp
if (a) {
} else if (b) {
} else if (c) {
}
```

Comment (Akkadius) [reply zone/bot_database.cpp:709]: Chances are else can just be the default and else is not usually necessary

Comment (Aeadoin) [reply zone/bot_database.cpp]: cleaned up, thanks!

**Review by Kinglykrab — APPROVED** [review]

**Pull request merged**